### PR TITLE
Do not run code style analyzers on generated files.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/12044

### CODE REVIEW

File: src/Features/CSharp/Portable/CodeFixes/RemoveUnnecessaryUsings/RemoveUnnecessaryUsingsDiagnosticProvider.cs [hunk 1]
```diff
@@ -1,106 +0,0 @@
-// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
-
-using System;
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeFixes/RemoveUnnecessaryUsings/RemoveUnnecessaryUsingsDiagnosticProvider.cs]: This was code that was not in our solution, but was still on the FS.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -40,10 +40,7 @@ internal sealed class CSharpAddBracesDiagnosticAnalyzer : DiagnosticAnalyzer, IB
                 SyntaxKind.LockStatement);
 
         public override void Initialize(AnalysisContext context)
```
Comment (dpoeschl) [inline src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs]: Should this be made a `CodeStyleAnalyzer` and override InitializeWorker instead?

File: src/Features/Core/Portable/CodeStyle/AbstractCodeStyleDiagnosticAnalyzer.cs [hunk 3]
```diff
@@ -69,5 +69,16 @@ private DiagnosticDescriptor CreateDescriptorWithId(string id, LocalizableString
                 severity,
                 isEnabledByDefault: true,
                 customTags: customTags);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            // Code style analyzers should not run on generated code.
```
Comment (dpoeschl) [inline src/Features/Core/Portable/CodeStyle/AbstractCodeStyleDiagnosticAnalyzer.cs]: Comment feels superfluous.

File: src/Features/Core/Portable/PopulateSwitch/PopulateSwitchDiagnosticAnalyzer.cs [hunk 4]
```diff
@@ -28,13 +28,14 @@ internal sealed class PopulateSwitchDiagnosticAnalyzer : DiagnosticAnalyzer, IBu
...
 
-        public override void Initialize(AnalysisContext context)
-        {
-            context.EnableConcurrentExecution();
+        private static MethodInfo s_registerMethod = typeof(AnalysisContext).GetTypeInfo().GetDeclaredMethod("RegisterOperationActionImmutableArrayInternal");
 
-            var internalMethod = typeof(AnalysisContext).GetTypeInfo().GetDeclaredMethod("RegisterOperationActionImmutableArrayInternal");
-            internalMethod.Invoke(context, new object[] { new Action<OperationAnalysisContext>(AnalyzeOperation), ImmutableArray.Create(OperationKind.SwitchStatement) });
-        }
+        public override void Initialize(AnalysisContext context)
```
Comment (dpoeschl) [inline src/Features/Core/Portable/PopulateSwitch/PopulateSwitchDiagnosticAnalyzer.cs]: Should this be a CodeStyleAnalyzer instead?

File: src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs [hunk 5]
```diff
@@ -37,19 +37,16 @@ internal abstract class AbstractUseAutoPropertyAnalyzer<TPropertyDeclaration, TF
         protected abstract SyntaxNode GetNodeToFade(TFieldDeclaration fieldDeclaration, TVariableDeclarator variableDeclarator);
 
         public sealed override void Initialize(AnalysisContext context)
```
Comment (dpoeschl) [inline src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs]: Should this be a CodeStyleAnalyzer instead?

**Comment by CyrusNajmabadi** [comment]
I think tehy should all be CodeStyleAnalzyers.  Are you ok with this if i make that change?

File: src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs]: Yup

File: src/Features/Core/Portable/PopulateSwitch/PopulateSwitchDiagnosticAnalyzer.cs (hunk 4 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/PopulateSwitch/PopulateSwitchDiagnosticAnalyzer.cs]: Yup

**Comment by CyrusNajmabadi** [comment]
@dpoeschl Have updated . Can you look?

File: src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs [hunk 6]
```diff
@@ -4,29 +4,34 @@
...
+        protected AbstractUseAutoPropertyAnalyzer()
+            : base(IDEDiagnosticIds.UseAutoPropertyDiagnosticId, s_title, s_title)
+        {
+        }
 
-        private readonly static DiagnosticDescriptor s_fadedTokenDescriptor = new DiagnosticDescriptor(
-            UseAutoPropertyFadedToken, FeaturesResources.Use_auto_property, FeaturesResources.Use_auto_property,
-            "Language", DiagnosticSeverity.Hidden, isEnabledByDefault: true,
-            customTags: new[] { WellKnownDiagnosticTags.Unnecessary });
+        //private readonly static DiagnosticDescriptor s_descriptor = new DiagnosticDescriptor(
```
Comment (dpoeschl) [inline src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs]: Delete instead of commenting out.

**Comment by CyrusNajmabadi** [comment]
Thanks.  HAve cleaned it up.

**Pull request merged**